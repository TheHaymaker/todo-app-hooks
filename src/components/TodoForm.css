input[type='text'] {
  border: none;
  box-shadow: 0px 2px 10px -5px black;
  background-color: #fff;
  line-height: 25px;
  font-size: 20px;
  width: 100%;
  display: inline-block;
  padding: 10px 15px;
}
