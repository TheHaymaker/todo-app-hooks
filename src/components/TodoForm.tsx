import React, { useState } from 'react';
import './TodoForm.css';

type todoInfo = { todos: Array<todoType>; saveTodos: Function };
type todoType = {
  name: string;
  completed: boolean;
};

const TodoForm = ({ todos, saveTodos }: todoInfo) => {
  const [value, setValue] = useState('');
  return (
    <form
      onSubmit={e => {
        e.preventDefault();
        saveTodos([...todos, { name: value, completed: false }]);

        setValue('');
      }}
    >
      <input
        placeholder="Add Todo"
        onChange={e => setValue(e.target.value)}
        type="text"
        value={value}
      />
    </form>
  );
};

export default TodoForm;
