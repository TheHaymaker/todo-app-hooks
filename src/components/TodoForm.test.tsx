import React from 'react';
import renderer from 'react-test-renderer';
import TodoForm from './TodoForm';

it('renders correctly', () => {
  const tree = renderer.create(<TodoForm todos={[]} saveTodos={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
