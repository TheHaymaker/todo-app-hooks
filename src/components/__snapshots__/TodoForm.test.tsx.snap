// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add Todo"
    type="text"
    value=""
  />
</form>
`;
